### ISSUE TITLE
can't access "this" property 

### ISSUE THREAD

**Issue by 75lb** [wycats/handlebars.js#967]
i'm trying and failing to access the `this` property in my context.. my context object looks something like:
```js
{  this: "whatever" }
```
but i can't access the value `"whatever"`.. i have tried all these template variations: 
```hbs
{{this.this}}
```
```hbs
{{this/this}}
```
```hbs
{{this.[this]}}
```
or is there another way?

**Comment by waynedpj** [comment]
i am no expert, but i believe `this` is a reserved word in JavaScript and [has a special meaning](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this).  try using another key for that property.

**Comment by 75lb** [comment]
`this` is a reserved word, the string value `"this"` is not.. it is a perfectly valid property name, just as `"return"`, `"if"`, `"break"` etc. are all valid property names.

**Comment by kpdecker** [comment]
This is a bug. It should resolve the property if using `{{[this]}}` or `{{this.this}}` syntax but something in the parser treats any references to this as a scope lookup.

```
./print-script "{{[this]}}"

Source:
1 '{{[this]}}'

Generated:
{"compiler":[6,">= 2.0.0-beta.1"],"main":function(depth0,helpers,partials,data) {
    return this.escapeExpression(this.lambda(depth0, depth0));
},"useData":true}

Map:
{"version":3,"sources":["input.hbs"],"names":[],"mappings":"AAAA,yCAAE;AAAA,IAAF,0DAAE;AAAA,CAAF","file":"output.js"}
```

**Comment by kpdecker** [comment]
Released in 3.0.2
